packages/pwa/app/utils/utils.js

```diff
@@ -158,7 +160,86 @@ export const capitalize = (text) => {
 }
 
 /**
- * A util to return current url configuration
- * @returns {object} - url object from the pwa-kit.config.json file
+ * This function return the identifiers (site and locale) from the given url
+ * The site will always go before locale if both of them are presented in the pathname
+ * @param path {string}
+ * @returns {object}
  */
-export const getUrlConfig = () => getConfig().app.url
+export const getParamsFromPath = (path) => {
+    const {pathname, search} = new URL(pathToUrl(path))
+
+    const config = getConfig()
+    const {pathMatcher, searchMatcherForSite, searchMatcherForLocale} = getConfigMatcher(config)
+    const pathMatch = pathname.match(pathMatcher)
+    const searchMatchForSite = search.match(searchMatcherForSite)
+    const searchMatchForLocale = search.match(searchMatcherForLocale)
+
+    // the value can only either in the path or search query param, there will be no overridden
+    const site = pathMatch?.groups.site || searchMatchForSite?.groups.site
+
+    const locale = pathMatch?.groups.locale || searchMatchForLocale?.groups.locale
+    return {site, locale}
+}
+
+/**
+ * This function returns the url config from the current configuration
+ * @return {object} - url config
+ */
+export const getUrlConfig = () => {
+    const {app} = getConfig()
+    if (!app.url) {
+        throw new Error("Can't find any valid url config. Please check your configuration file.")
+    }
+    return app.url
+}
+
+/**
+ * This function return the regex for matching site and locale
+ * @param config
+ * @return {{searchMatcherForSite: RegExp, searchMatcherForLocale: RegExp, pathMatcher: RegExp}}
+ */
+export const getConfigMatcher = (config) => {
+    if (!config) {
+        throw new Error('Config is not defined.')
+    }
+
+    const allSites = getSites()
+    let sites = []
+    let locales = []
+    allSites.forEach((site) => {
+        sites.push(site.alias)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            locales.push(locale.id)
+        })
+    })
+
+    allSites.forEach((site) => {
+        sites.push(site.id)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            locales.push(locale.alias)
+        })
+    })
+
+    // filter out falsy values
+    locales = locales.filter(Boolean)
+    sites = sites.filter(Boolean)
+    // prettier-ignore
+    // eslint-disable-next-line
+    const searchPatternForSite = `site=(?<site>${sites.join('|')})`
+    // prettier-ignore
+    // eslint-disable-next-line
+    const pathPattern = `(?:\/(?<site>${sites.join('|')}))*(?:\/(?<locale>${locales.join("|")}))*(?!\\w)`
+    // prettier-ignore
+    // eslint-disable-next-line
+    const searchPatternForLocale = `locale=(?<locale>${locales.join('|')})`
+    const pathMatcher = new RegExp(pathPattern)
+    const searchMatcherForSite = new RegExp(searchPatternForSite)
+    const searchMatcherForLocale = new RegExp(searchPatternForLocale)
```

Comment:
I noticed that you are using 2 regular expressions here. I believe that we talked about this before with regards to whether or not we can create a regex that will ignore order. Anyway. I was able to create one that ignore order so you could potentially combine them. 

Here is my example:

```
var regex = /(?=.*site=(?<site>us|global))(?=.*locale=(?<locale>en|fr))/

"site=us&locale=fr".match(regex) // Captures 2 groups, locale = fr and site = us

"locale=fr&site=us".match(regex) // Captures 2 groups, locale = fr and site = us

"locale=fr&something=else&site=us".match(regex) // Captures 2 groups, locale = fr and site = us
```

Note I didn't spend a lot of time on this, so you'd have to validate it before using it.

Reply:
Thanks for the feedback, I will test it out 👍

Reply:
I tried some more different cases that have only of the group, it did not work
```
"site=us" => return null
"locale=fr" => return null
```
any idea how to make those work? 🤔